auto_archive.py

```diff
@@ -58,169 +44,101 @@ def batch_if_valid(col, val, final_value=None):
-                        exit()
-                    except Exception as e:
-                        result = False
-                        logger.error(f'Got unexpected error in row {row} with archiver {archiver} for url {url}: {e}\n{traceback.format_exc()}')
-
-                    if result:
-                        # IA is a Success I believe - or do we want to display a logger warning for it?
-                        if result.status in ['success', 'already archived', 'Internet Archive fallback']:
-                            result.status = archiver.name + \
-                                ": " + str(result.status)
-                            logger.success(
-                                f'{archiver} succeeded on row {row}, url {url}')
-                            break
-
-                        # wayback has seen this url before so keep existing status
-                        if "wayback: Internet Archive fallback" in result.status:
-                            logger.success(
-                                f'wayback has seen this url before so keep existing status on row {row}')
-                            result.status = result.status.replace(' (duplicate)', '')
-                            result.status = str(result.status) + " (duplicate)"
-                            break
-
-                        logger.warning(
-                            f'{archiver} did not succeed on {row=}, final status: {result.status}')
-                        result.status = archiver.name + \
-                            ": " + str(result.status)
-                # get rid of driver so can reload on next row
-                driver.quit()
+            is_retry = False
+            if url == '' or status not in ['', None]:
+                is_retry = Archiver.should_retry_from_status(status)
+                if not is_retry: continue
+
+            # All checks done - archival process starts here
+            gw.set_cell(row, 'status', 'Archive in progress')
+            url = expand_url(url)
+            c.set_folder(gw.get_cell_or_default(row, 'folder', default_folder, when_empty_use_default=True))
+
+            # make a new driver so each spreadsheet row is idempotent
+            c.recreate_webdriver()
+
+            # order matters, first to succeed excludes remaining
+            active_archivers = [
+                TelethonArchiver(storage, c.webdriver, c.telegram_config),
+                TiktokArchiver(storage, c.webdriver),
+                YoutubeDLArchiver(storage, c.webdriver, c.facebook_cookie),
+                TelegramArchiver(storage, c.webdriver),
+                TwitterArchiver(storage, c.webdriver),
+                WaybackArchiver(storage, c.webdriver, c.wayback_config)
+            ]
+
+            for archiver in active_archivers:
+                logger.debug(f'Trying {archiver} on {row=}')
+
+                try:
+                    result = archiver.download(url, check_if_exists=True)
```

Comment:
`check_if_exists` could be a configurable option too

Reply:
put into bool config `--check-if-exists`